graylog-labs/graylog-plugin-aws
# Refactor CloudWatch related code and add logs input

- Refactor CloudWatch related code
  - Create a generic Kinesis transport and reader that can be used by different inputs
  - Create an abstract CloudWatch log data codec
  - Rename some flow log classes to more generic names
- Update to the latest AWS SDK and Kinesis client
- Use unique application name per stream to avoid state problems in DynamoDB
- Periodically checkpoint the sequence number to make sure the consumer continues where it left off instead of only fetching new data after a Graylog server restart
- Use auth provider in InstanceLookupTable so it also works with instance profiles, environment variables, etc
- Inject InstanceLookupTable and bind it as singleton
